src/components/chip-set/chip-set.tsx

```diff
@@ -466,6 +473,15 @@ export class ChipSet {
     }
 
     private isInvalid() {
+        if (this.readonly) {
+            // A readonly field can never be invalid.
+            return false;
+        }
+
+        if (this.invalid) {
+            return true;
+        }
+
```

Comment:
if you wonder about why, it could be good to look at:

https://github.com/Lundalogik/lime-elements/blob/5744bb981d253866cdc7553fb3f6e01ab2ae3296/src/components/input-field/input-field.tsx#L544C9-L544C9

Reply:
I still don't understand why we need a special case for readonly. It would be weird to show a readonly field as invalid to the user, because they can't change it to be valid, but technically it seems possible to me. Readonly and invalid would indicate something that's outside the control of the user is in an inconsistent state.

Reply:
a `readonly` field is a stupid/smart internal notion that we have invented, due to CRM. An input field which is inherently interactive cannot be `readonly`. It doesn't make sense. A readonly field is basically something that is for visualization of data. That's it. It's a dead component, non-interactive. 

So when something is non-interactive and dead, the system cannot tell the user to fix it, because it is now `invalid`! Users cannot do anything about the validity of the field. 😅 

Hope I could explain this in short @vicgeralds

Reply:
> So when something is non-interactive and dead, the system cannot tell the user to fix it, because it is now invalid! Users cannot do anything about the validity of the field. 😅

Yes, sounds reasonable. That still doesn't fully motivate the special case inside `isInvalid()` though. 🙂 

Let me put it this way: would it make any difference if we removed these lines?

```js
        if (this.readonly) {
            // A readonly field can never be invalid.
            return false;
        }
```

If that would mess up styling, or causes some other problem when rendering, fine. I think I would still prefer just handling that in css if possible. 🙂

Reply:
Just some food for thought:
Lets say the customers have logic on their deals. Any deal in the segment X, needs to have a `Total Value` higher than `100 000`. The `Total Value` is never set by the user, it is calculated from the `Price per Unit` x `Amount`. So once the user tries to save the custom validation logic stops them. 

Which field should we then show as the invalid one? In my opinion it can only be the `Total Value` field.

Reply:
> Which field should we then show as the invalid one? In my opinion it can only be the `Total Value` field.

In other words, a readonly field. Good point!

Another approach is to make the section of group of fields that has an error appear as invalid. But currently, a validation error has to belong to a field, from what I understand, and a limeobject is just a flat collection of fields.

Reply:
Agree, and you could in this case raise the errors on both the `Price per Unit` and `Amount`, but that is not always intuitive for the user.

Reply:
I understand what you mean. 

I don't want to be religious about it, but I still disagree (or agree with what I said 😅 ).

But for interactive components (e.g. input elements) `invalid` is a system/developer thing really, which also manifests itself to the user. It is **NEVER** good enough however, because merely by making the field 🔴 the UI will be unable to communicate to the user "what is exactly wrong with this field?". 

In a good design, the system / developer / designer will use a proper helper text to explain what is wrong. This helper text is not the helpertext that usually is provided with the input field. It is rather the description of the problem. Are we doing that now anywhere in the web client? I don't think so.

But that is when the field is interactive. The invalid state can become valid, by interacting with the field **_itself_**!!!! Not by doing something else somewhere else in the UI.

For what you explained @simonedeblasio , the filed is non-interactive and it is just a visualization of data. So we can't use the same convention and make it "invalid" and expect the user to fix it. The user cannot do anything about that particular field itself, and their attempts of fixing the problem should be in other input field. 

I would argue that the problem that you describe has a complexity that perhaps needs a set of UI changes, to communicate or guide the user about how they can fix the invalid state. Maybe by showing a snackbar or modal with a more clear and more elaborated message for the user, or by setting some sort of highlight on all fields that are involved in the validation. 

In your case, the `Price per Unit` & `Amount` are the problems, not the `total value`. 

Can the total value become 🔴 as well? Well, yes, maybe. 
Is it enough for the user to know what to do? I don't think so. But maybe?
Are there better ways? I think so.
Should we allow readonly fields to become red and show a validation error? I'm not convinced yet, but maybe.

This is how far my rationale reaches right now. Open for suggestions or counter arguments.